jobs-service/kogito-addons-jobs-service/kogito-addons-quarkus-jobs-service-embedded/deployment/src/main/java/org/kie/kogito/addons/quarkus/jobs/service/embedded/deployment/KogitoAddonsQuarkusJobsServiceEmbeddedProcessor.java

```
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.kogito.addons.quarkus.jobs.service.embedded.deployment;
+
+import org.kie.kogito.quarkus.addons.common.deployment.KogitoCapability;
+import org.kie.kogito.quarkus.addons.common.deployment.OneOfCapabilityKogitoAddOnProcessor;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.SystemPropertyBuildItem;
+
+class KogitoAddonsQuarkusJobsServiceEmbeddedProcessor extends OneOfCapabilityKogitoAddOnProcessor {
+
+    private static final String FEATURE = "kogito-addons-quarkus-jobs-service-embedded";
+    private static final String JOBS_SERVICE_URL = "kogito.jobs-service.url";
+    private static final String SERVICE_URL = "kogito.service.url";
+
+    KogitoAddonsQuarkusJobsServiceEmbeddedProcessor() {
+        super(KogitoCapability.SERVERLESS_WORKFLOW, KogitoCapability.PROCESSES);
+    }
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void buildConfiguration(BuildProducer<SystemPropertyBuildItem> systemProperties) {
+        systemProperties.produce(new SystemPropertyBuildItem(SERVICE_URL, "http://${quarkus.http.host}:${quarkus.http.port}"));
```

Not sure if it possible, and maybe this can be done in a separate JIRA..., but if this add-on is somehow forcing work hand in hand with jobs-service-management-addon, at build time, if we detect that any of the other addons is present, we can throw an error or a warning.

Following the same reasoning, I think that (maybe by using simple runtime class on this addon) we can write a warning meesage in the log every time we start the application with this addon saying that it's intended for development/tests purposes.
I think this information might help users in the future, considering that when the project is finally let's say moved to a "production" project, users will have to manually remove this dependency from the project. So if we can add such a message we can help users to don't forget this "cleaning" task.

---

Yeah, the idea to validate and check if there is another jobs addon is great, in fact we need to do this for every addon that might have different options, like persistence and even data-index. 
I'll see what we can do, but probably it is better to open a new Jira for it.